src/scripts/migrate.py

```diff
@@ -0,0 +1,139 @@
+    'mondo_id': 'ID', 'xref': 'A oboInOwl:hasDbXref', 'label': 'LABEL', 'definition': 'A IAO:0000115',
+    'parents': 'SC %'}
+
+
+def slurp(
+    ontology_path: str, onto_config_path: str, sssom_map_path: str, min_id: int, max_id: int, mondo_terms_path: str,
+    slurp_dir_path: str, outpath: str, use_cache=False
+) -> pd.DataFrame:
+    """Run slurp pipeline for given ontology
+    todo: tried on an older computer and it was indeed too slow. Has to do w/ term class / utils
+    """
+    # Read inputs
+    ontology: ProntoImplementation = _load_ontology(ontology_path, use_cache)
+    with open(onto_config_path, 'r') as stream:
+        onto_config = yaml.safe_load(stream)
+        owned_prefix_map: Dict[PREFIX, URI] = onto_config['base_prefix_map']
+    sssom_df: pd.DataFrame = pd.read_csv(sssom_map_path, comment='#', sep='\t')
+    mondo_term_ids: Set[int] = get_mondo_term_ids(mondo_terms_path, slurp_dir_path)
+
+    # Intermediates
+    owned_terms: List[Term] = _get_all_owned_terms(
+        ontology=ontology, owned_prefix_map=owned_prefix_map, ontology_path=ontology_path, cache_dir_path=CACHE_DIR,
+        onto_config_path=onto_config_path, use_cache=use_cache)
+    owned_term_curies: List[CURIE] = [x.curie for x in owned_terms]
+    sssom_object_ids: Set[Union[URI, CURIE]] = set(sssom_df['object_id'])  # Usually CURIE, but spec allows URI
+    unmapped_terms: List[Term] = [x for x in owned_terms if x.curie not in sssom_object_ids]
+
+    # todo: after sparql: temp: check % that have parents
+    import os
+    ratio = len([x for x in unmapped_terms if x.direct_owned_parent_curies]) / len(unmapped_terms)
+    print(f'{os.path.basename(ontology_path)}: % terms that "have parents": {int(ratio) * 100}%')
+
+    # Determine slurpable terms
+    # todo: check. does this code meet requirement for 'slurpable term' below?
+    # - Slurpable term: A term for which all parents are already slurped. In other words, no parent is owned by the
+    # ontology but unmapped in Mondo.
+    next_mondo_id = min_id
+    terms_to_slurp: List[Dict[str, str]] = []
+    for t in unmapped_terms:
+        # If all T.parents mapped, and at least one of them is an exact or narrow match, designate T for slurping
+        # (i.e. only slurp if parents are already slurped)
+        qualified_parents = True
+        parent_mondo_ids = []
+        # TODO: are these conditions correct? We're getting no outputs, though many terms w/ parents exist
+        for parent_curie in t.direct_owned_parent_curies:
+            # Conversely, if any of T.parents is unmapped, T is not to be slurped
+            if parent_curie in sssom_object_ids:
+                parent_df: pd.DataFrame = sssom_df[sssom_df['object_id'] == parent_curie]
+                parent_mondo_id = parent_df.to_dict('records')[0]['subject_id']
+                parent_mondo_ids.append(parent_mondo_id)
+            if parent_curie not in sssom_object_ids and parent_curie in owned_term_curies:
```

Comment:
## Not getting any outputs
I was getting `qualified_parents = True` for all terms at first, so my outputs were very large.

With not much debugging, I figured out why that was and fixed it.

In this block, I needed to change `owned_terms` to `owned_term_curies`.
```py
if parent_curie not in sssom_object_ids and parent_curie in owned_term_curies:
    qualified_parents = False
    break
```

The `owned_terms` was a list of my `Term` objects, not curies. That's why this condition was never being met. However after fixing it, I'm no longer getting any outputs. It seems that, for right now, for all ontologies we're slurping, for all terms in all those ontologies, there is at least 1 parent which has not been mapped, and thus we cannot slurp.

This is strange, though. If there are unmapped parents of a term, then those unmapped parents should be slurp candidates as well. So I would expect to at least see some terms higher up in the hierarchy that qualify as slurpable...

I can try debugging this further, but if I can get a concrete example of a simple term that is slurpable, that would really help. The example you gave me `DOID:0060091` didn't work because it is already mapped. I can probably manually look through the inputs and try to find an example myself, but it would save me a lot of time if you happen to know one that should work.

Reply:
Here is how you find a good candidate:

1. Print the unmapped terms
2. Pick a random one (T)
3. Look at the _direct_ parents P of that term T (in DOID, nearly all terms will have only a single parent)
4. If the P is in `sssom_object_ids`,  T should be slurped

The vast majority of unmapped terms should fulfill that criterion!

Reply:
Make sure the parent method only looks at direct isa-parents, not _all_ parents (non-isa, or indirect isa, i.e. grandparents)

Reply:
Hmm, I do believe that (1-4) is what the algorithm is doing explicitly. But I will do this a bit manually and see if I find an error in my logic, or otherwise find some good concrete examples to test. I guess I could make a unit test using some smaller subsets of the files (e.g. a `doid.owl` with just a few classes; a SSSOM file with just a few rows, etc).

In regards to which types of parents, there are some OAK methods that get only subclass/isa relationships. Still somewhat broken, so just using SPARQL now, and my query is only getting `rdfs:subClassOf*`.

Reply:
Solved via `rdfs:subClassOf*` -> `rdfs:subClassOf`